packages/artifact_proxy/lib/artifact_proxy.dart

```diff
@@ -0,0 +1,87 @@
+// ignore_for_file: avoid_print
+
+import 'package:collection/collection.dart';
+import 'package:shelf/shelf.dart';
+
+/// A [Handler] that proxies artifact requests to the correct location.
+/// This is determined based on the [config].
+Handler artifactProxyHandler({required Map<dynamic, dynamic> config}) {
+  final engineMappings = config['engine_mappings'] as Map;
+  final shorebirdEngineRevisions = engineMappings.keys.cast<String>();
+
+  return (Request request) {
+    final path = request.url.path;
+    final shorebirdEngineRevision = shorebirdEngineRevisions.firstWhereOrNull(
+      path.contains,
```

Comment:
Do you want contains? or ends with?

Reply:
I think we still need contains because I don't think ends with is strict enough?